hs/src/Reach/Backend/JS.hs

```diff
@@ -324,6 +324,28 @@ jsArg_m = \case
   Nothing -> return $ "undefined /* Nothing */"
   Just a -> jsArg a
 
+shouldHashMapKey :: DLType -> Bool
+shouldHashMapKey = \case
+  T_Null -> False
+  T_Bool -> False
+  T_UInt {} -> False
+  T_Digest -> False
+  T_Address -> False
+  T_Contract -> False
+  T_Token -> False
+  T_Bytes {} -> True
+  T_Array {} -> True
+  T_Data {} -> True
+  T_Tuple {} -> True
+  T_Object {} -> True
+  T_Struct {} -> True
+
+jsMapKey :: DLArg -> App Doc
+jsMapKey k =
+  case shouldHashMapKey $ argTypeOf k of
```

Comment:
I would consider making this just always true... why would it be bad to always use digests?

Reply:
Map lookups would fail because the connectors only hash keys that containers or strings